homework/index.js

```diff
@@ -30,14 +30,120 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const listContainer = document.getElementById('list-container');
+    clearContainer(listContainer);
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert alert-error' });
+  }
+
+  function addRow(tbody, label, repository) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    if (label === 'Repository:') {
+      const urlRepo = repository.html_url;
+      const td = createAndAppend('td', tr);
+      createAndAppend('a', td, {
+        text: repository.name,
+        href: urlRepo,
+        target: '_blank',
+      });
+    }
+    if (label === 'Description:') {
+      createAndAppend('td', tr, { text: repository.description });
+      console.log(repository.description);
+    }
+    if (label === 'Forks:') {
+      createAndAppend('td', tr, { text: repository.forks });
+    }
+
+    if (label === 'Updated:') {
+      createAndAppend('td', tr, { text: repository.updated_at });
+    }
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const div = createAndAppend('div', container, { id: 'info_div' });
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    createAndAppend('tr', table);
+    createAndAppend('tr', table, {});
+    addRow(tbody, 'Repository:', repository);
+    addRow(tbody, 'Description:', repository);
+    addRow(tbody, 'Forks:', repository);
+    addRow(tbody, 'Updated:', repository);
+  }
+
+  function createContributorsList(repositoryName, ul) {
+    clearContainer(ul);
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}/contributors`;
+    fetchJSON(url, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+
+      createAndAppend('h2', ul, {
```

Comment:
Here is some invalid HTML too. As I said, a `ul` can only have `li` children, and note an `h2` as you have here:

```html
<ul id="Contributors_list">
  <h2>contributions</h2>
  <li>
    <div>
      <img
        class="contributor_avatar"
        src="https://avatars0.githubusercontent.com/u/15912395?v=4"
        alt="gijscor"
      />
      <p id="p1">gijscor</p>
      <p id="p2">47</p>
    </div>
  </li>
  <li>
    <div>
      <img
        class="contributor_avatar"
        src="https://avatars1.githubusercontent.com/u/37488847?v=4"
        alt="wouterkleijn"
      />
      <p id="p1">wouterkleijn</p>
      <p id="p2">11</p>
    </div>
  </li>
</ul>
```